### PR TITLE
Add desktop application canvas

### PR DESCRIPTION
**Pull request by sbrunner**
Create apisrc folder
Export desktop alt as a lib
Add some documentation

### CODE REVIEW

File: src/canvas/DesktopElement.ts [hunk 1]
```diff
@@ -325,10 +319,20 @@ export default class GmfDesktopCanvas extends BaseElement {
...
   }
 
   toggleShowInfobar_(): void {
     this.showInfobar_ = !this.showInfobar_;
+    this.toggleBackgroundSelector();
+  }
+
+  toggleBackgroundSelector(): void {
+    if (this.showInfobar_) {
+      document.documentElement.style.setProperty(`--current-footer-height`, 'var(--infobar-height)');
```
Comment (sbrunner) [inline src/canvas/DesktopElement.ts:332]: This is working?

File: contribs/gmf/src/controllers/desktop.scss [hunk 2]
```diff
@@ -790,7 +790,7 @@ gmf-featurestyle {
   }
```
Comment (sbrunner) [inline contribs/gmf/src/controllers/desktop.scss:790]: Should be removed :-)

File: src/canvas/DesktopElement.ts (hunk 1 above)
Comment (sbrunner) [reply src/canvas/DesktopElement.ts:332]: Nice :-)

File: contribs/gmf/src/controllers/desktop.scss (hunk 2 above)
Comment (llienher) [reply contribs/gmf/src/controllers/desktop.scss:790]: Is it not used in the non-canvas interface (desktop interface) ?

Comment (sbrunner) [reply contribs/gmf/src/controllers/desktop.scss:790]: Oups your right :-)

File: contribs/gmf/apps/desktop/index.html.ejs [hunk 3]
```diff
@@ -104,12 +104,18 @@
...
-          ng-prop-login_info_message="mainCtrl.loginInfoMessage"
-          ng-prop-post_loading="mainCtrl.postLoading"
-          ng-prop-password_validator="mainCtrl.passwordValidator"
-          ></gmf-auth-panel>
+            <div class="row"
+                ng-show="mainCtrl.loginActive">
+              <div class="col-sm-12">
+                <a class="btn close gmf-close" ng-click="mainCtrl.loginActive = false">&times;</a>
+                <gmf-auth-panel ng-show="mainCtrl.loginActive"
+                  ng-on-close-panel="mainCtrl.loginActive = $event.detail"
```
Comment (llienher) [inline contribs/gmf/apps/desktop/index.html.ejs:112]: ng-on is not usefull if the close button is now back in the index. :-)

File: docs/developer-guide.md [hunk 4]
```diff
@@ -6,6 +6,14 @@ This document provides information for developers working on ngeo.
 
 Type `make help` to display available targets.
 
+## Code structure
+
+We have 3 repository with sources code:
+
+- `src`: this folder contain the major part of source code.
+- `srcapi`: this folder contain the source code witch we offer a more stable API.
+- `api`: this folder contain the JavaScript api of the project.
```
Comment (llienher) [inline docs/developer-guide.md:15]:
```suggestion
- `src`: this folder contains the major part of source code.
- `srcapi`: this folder contains the source code which we offer a more stable API.
- `api`: this folder contains the JavaScript API of the project.
```

File: docs/developer-guide.md [hunk 5]
```diff
@@ -6,6 +6,14 @@ This document provides information for developers working on ngeo.
 
 Type `make help` to display available targets.
 
+## Code structure
+
+We have 3 repository with sources code:
```
Comment (llienher) [inline docs/developer-guide.md:11]:
```suggestion
We have 3 repositories with sources code:
```

File: srcapi/store/panels.ts [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import {BehaviorSubject} from 'rxjs';
+
+/**
+ * A projection definitions.
+ */
+export type OpenPanelOptions = {
+  /**
+   * True to open the panel, false to cose, undefined to toggle.
```
Comment (llienher) [inline srcapi/store/panels.ts:29]:
```suggestion
   * True to open the panel, false to close, undefined to toggle.
```

File: srcapi/store/panels.ts [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+   */
+  state?: boolean;
+  /**
+   * Automatically close the panel on tool panel change, for footer panel.
+   */
+  autoClose?: boolean;
+  /**
+   * Don't throw en error on closing an unopened panel.
```
Comment (llienher) [inline srcapi/store/panels.ts:37]:
```suggestion
   * Don't throw an error on closing an unopened panel.
```

**Review by llienher — APPROVED** [review]
Just some typos, and a line to probably test and remove if needed (ng-on) in the index.

**Review by llienher — CHANGES_REQUESTED** [review]
Please check and fix the mobile and mobile_alt interface, it should use the form directly. Currently no form authentication is visible.

**Comment by sbrunner** [comment]
It's also broken in the master branch, then it can be fixed in another pull request...

**Review by llienher — APPROVED** [review]
LGTM !

**Comment by sbrunner** [comment]
Thanks @llienher :-)

**Pull request merged**